commercetools-internal-processors/src/main/java/io/sphere/sdk/annotations/processors/generators/BaseAbstractGenerator.java

```diff
@@ -237,26 +238,33 @@ protected AnnotationSpec createJsonPropertyAnnotation(final String jsonName) {
     }
 
     protected MethodSpec createFactoryMethod(final FactoryMethod factoryMethod, final List<PropertyGenModel> properties, final ClassName returnType) {
-        final Set<String> factoryParameterNames = Stream.of(factoryMethod.parameterNames()).collect(Collectors.toCollection(LinkedHashSet::new));
+        return createFactoryMethod(properties,returnType,factoryMethod.parameterNames(),factoryMethod.methodName(),factoryMethod.useLowercaseBooleans());
+    }
+
+    protected MethodSpec createFactoryMethod(final List<PropertyGenModel> properties, final ClassName returnType, final String[] parametersName,
+                                             final String methodName,final boolean useLowercaseBooleans) {
+        final Set<String> factoryParameterNames = Stream.of(parametersName).collect(Collectors.toCollection(LinkedHashSet::new));
         final Map<String, PropertyGenModel> getterMethodByPropertyName = properties.stream()
                 .collect(Collectors.toMap(PropertyGenModel::getName, Function.identity()));
         final List<PropertyGenModel> parameterTemplates = factoryParameterNames.stream()
                 .map(getterMethodByPropertyName::get)
+                .filter(Objects::nonNull)
```

Comment:
Actually this would help in case another property required in the factory method can not be found in the object , but i would agree using an assertion isn't quite required 😢 
But since we added the `messager ` now i can print a pretty message that shows what went wrong during processing of the annotation.